[MAC] - Light(Preview) theme - Fix for missing alternate row colors for Tree

mvm-sap opened the pull request:
Tree Viewers had alternate row colors if they had coloumns, in Light Theme. This was missing in Light(Preview) theme. This has been fixed with this change by making swt-lines-visible (Tree Property) to true.
Please refer #2395

BeckerWdf commented on bundles/org.eclipse.ui.themes/css/e4_preview_mac.css:
```diff
@@ -126,6 +125,11 @@ CTabFolder Canvas {
 }
 
+.View Composite Tree{
+	background-color: #f8f8f8;
```
Tested on macOS.
Zebra-Lines are not visible because of this line. If this is removed. Zebar-Lines are visible.

BeckerWdf replied:
```suggestion
```

BeckerWdf reviewed (commented):
looks good

BeckerWdf commented:
with this PR:
<img width="1072" alt="Screenshot 2024-10-15 at 16 45 24" src="https://github.com/user-attachments/assets/fe0d64e4-a828-401a-8d04-c54e62341232">

before:
<img width="1072" alt="Screenshot 2024-10-15 at 16 46 48" src="https://github.com/user-attachments/assets/599aaee3-381f-4ba0-91cd-14f6a9c77341">

mvm-sap commented:
> with this PR: <img alt="Screenshot 2024-10-15 at 16 45 24" width="1072" src="https://private-user-images.githubusercontent.com/28338612/376666467-fe0d64e4-a828-401a-8d04-c54e62341232.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkwNTA4NjksIm5iZiI6MTcyOTA1MDU2OSwicGF0aCI6Ii8yODMzODYxMi8zNzY2NjY0NjctZmUwZDY0ZTQtYTgyOC00MDFhLThkMDQtYzU0ZTYyMzQxMjMyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE2VDAzNDkyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQyNzk3YmYwNDhkMDkzODgxNmNlYWMxYzYwZjdmZjU5Y2Q3NWZkM2JjODdhYzBjNmFkNDEyOTQ1ODYxYjllNDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.tIJsjuMkyjf6yrlNUsR60MrOL-XLOyrP5XxO0Qi_MtU">
> 
> before: <img alt="Screenshot 2024-10-15 at 16 46 48" width="1072" src="https://private-user-images.githubusercontent.com/28338612/376666395-599aaee3-381f-4ba0-91cd-14f6a9c77341.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkwNTA4NjksIm5iZiI6MTcyOTA1MDU2OSwicGF0aCI6Ii8yODMzODYxMi8zNzY2NjYzOTUtNTk5YWFlZTMtMzgxZi00YmEwLTkxY2QtMTRmNmE5Yzc3MzQxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE2VDAzNDkyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ3YjBiM2Y1OTFkYTg1YzllM2YwOWQyZWUxMjhiOGQxNjg3N2M2YmQwZThiNWU2YTc5NjY4ZmUyN2JjZTRiNjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.rjQDa3yE4PXvpAVhzjhZo5_EuM1wcdPnU-DW54oqN3g">

In after image, Project Explorer and Outline view have white background

BeckerWdf commented:
> In after image, Project Explorer and Outline view have white background

You are right. So they requirement would be: Show the zebra-lines in Trees *but only* when the tree has more then one column. Correct?

mvm-sap commented:
> > In after image, Project Explorer and Outline view have white background
> 
> You are right. So they requirement would be: Show the zebra-lines in Trees _but only_ when the tree has more then one column. Correct?

Yes

Phillipus commented:
> So they requirement would be: Show the zebra-lines in Trees _but only_ when the tree has more then one column. Correct?

Actually zebra lines are only visible when `Tree#setLinesVisible(true)`

BeckerWdf commented on bundles/org.eclipse.ui.themes/css/e4_preview_mac.css:
```diff
@@ -195,6 +194,10 @@ CTabFolder Canvas {
 	background-color: #ffffff;
 }
 
+#org-eclipse-ui-navigator-ProjectExplorer Tree{
```
We should reference concrete views of the Eclipse SDK.

BeckerWdf commented:
Do it get it right: Mac creates these zebra lines for trees only if they have more then two "columns"? Is that a feature of the native UI toolkit or is this controlled by the SWT implementation?

Is it correct that setting background color does overwrite the "show visible lines" setting? Is that think that is controlled by the SWT implementation?

Do we have a possibility to tell SWT so set the background only if "show visible lines" is false / if the tree as more then two "columns"?

Phillipus commented:
> Mac creates these zebra lines for trees only if they have more then two "columns"? 

No. Zebra lines are only visible when `Tree#setLinesVisible(true)` with any amount of columns (or none).

mvm-sap commented:
> > Mac creates these zebra lines for trees only if they have more then two "columns"?
> 
> No. Zebra lines are only visible when `Tree#setLinesVisible(true)` with any amount of columns (or none).

Below is my understanding

 **Light theme** 
 Project explorer, Outline View - Tree Control - has only one column - bg color is white - setLinesVisible is false
 Problems View  - Tree Control -  has more than one column - bg has alternate row colors - setLinesVisible is true

**Light (Preview) theme**
 we just set the bg color of Tree control to grey (#F8F8F8) in CSS file
```
.View Tree {
swt-background-color: #f8f8f8 
}
```

 Project explorer, Outline View - Tree Control - has only one column - bg color is grey - setLinesVisible is false
 Problems View  - Tree Control -  has more than one column - bg color is grey - **setLinesVisible is set to false automatically**

BeckerWdf committed:
Only set background color on tree if "linesVisible" is false

On macOS setting "linesVisible" to true creates an zebra styled pattern
on the tree. If we now set the background color via CSS
on this tree that pattern would be gone. Setting the background color
via CSS on a tree that has "linesVisible" to false does not do any harm.

BeckerWdf commented:
I just pushed a fix for this.
@Phillipus: Can you pls test this on your side?

Phillipus commented:
> I just pushed a fix for this.
@Phillipus: Can you pls test this on your side?

Tested, and I see zebras. 🦓 🦓 🦓 🦓 

Thanks!

mvm-sap commented:
> I just pushed a fix for this. @Phillipus: Can you pls test this on your side?

Looks like a good solution. Thanks, Matthias.
